vdp/pipeline/v1beta/pipeline.proto

```diff
@@ -1938,3 +1938,189 @@ message LookUpPipelineAdminResponse {
   // The requested pipeline.
   Pipeline pipeline = 1;
 }
+
+// ListPipelineRunsRequest is the request message for ListPipelineRuns.
+message ListPipelineRunsRequest {
+  // The namespace to list pipeline runs for.
+  string namespace = 1;
+
+  // The page number to retrieve (1-based).
+  int32 page = 2;
+
+  // The number of items per page.
+  int32 page_size = 3;
+}
+
+// ListPipelineRunsResponse is the response message for ListPipelineRuns.
+message ListPipelineRunsResponse {
+  // The list of pipeline runs.
+  repeated PipelineRun pipeline_runs = 1;
+
+  // The total number of pipeline runs matching the request.
+  int64 total_count = 2;
+
+  // The current page number.
+  int32 page = 3;
+
+  // The number of items per page.
+  int32 page_size = 4;
+}
+
+// ListComponentRunsRequest is the request message for ListComponentRuns.
+message ListComponentRunsRequest {
+  // The namespace of the pipeline run.
+  string namespace = 1;
+
+  // The unique identifier of the pipeline run to list component runs for.
+  string pipeline_trigger_uid = 2;
+
+  // The page number to retrieve (1-based).
+  int32 page = 3;
+
+  // The number of items per page.
+  int32 page_size = 4;
+}
+
+// ListComponentRunsResponse is the response message for ListComponentRuns.
+message ListComponentRunsResponse {
+  // The list of component runs.
+  repeated ComponentRun component_runs = 1;
+
+  // The total number of component runs matching the request.
+  int64 total_count = 2;
+
+  // The current page number.
+  int32 page = 3;
+
+  // The number of items per page.
+  int32 page_size = 4;
+}
+
+// FileReference represents metadata for a file.
+message FileReference {
+  // Name of the file.
+  string name = 1;
+
+  // Format of the file (e.g., PDF, TXT, JPG).
+  string type = 2;
+
+  // Size of the file in bytes.
+  int64 size = 3;
+
+  // URL of the file (e.g., S3 URL).
+  string url = 4;
+}
+
+// PipelineRun represents a single execution of a pipeline.
+message PipelineRun {
+  // Unique identifier for the pipeline.
+  string pipeline_uid = 1;
+
+  // Unique identifier for each run.
+  string pipeline_trigger_uid = 2;
```

Comment:
We should use either `trigger` or `run` but not both
```suggestion
  string pipeline_run_uid = 2;
```

@joremysh not a strong opinion but you called this `uid` in the model endpoint. I'm fine with either approach but I'd make it consistent across APIs.

Reply:
I could update the name in protobuf first and later for those in code that are not related to protogen-go